packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/Providers/ModelProviders/ModelProviderTests.cs

```diff
@@ -349,43 +349,58 @@ public void BuildModelAsStruct()
 
         [TestCase(true)]
         [TestCase(false)]
-        public void TestBuildFields(bool containsUnknownAdditionalProp)
+        public void TestBuildFields(bool containsMixedAdditionalProperties)
         {
-            InputType? additionalProperties = containsUnknownAdditionalProp ? InputPrimitiveType.Any : null;
+            InputType? additionalProperties = containsMixedAdditionalProperties
+                ? InputFactory.Union([InputPrimitiveType.Float64, InputPrimitiveType.Int64, InputPrimitiveType.String])
+                : null;
             var inputModel = InputFactory.Model("TestModel", properties: [], additionalProperties: additionalProperties);
             var modelTypeProvider = new ModelProvider(inputModel);
             var fields = modelTypeProvider.Fields;
 
-            // Assert
             Assert.IsNotNull(fields);
-            if (containsUnknownAdditionalProp)
+
+            if (containsMixedAdditionalProperties)
             {
-                Assert.AreEqual(0, fields.Count);
+                Assert.AreEqual(4, fields.Count);
+                Assert.AreEqual("_additionalBinaryDataProperties", fields[0].Name);
```

Comment:
I see a lot of asserts for the field names and param names, can we get a test which asserts the property names / accessors?

Reply:
sure, added.